packages/sui-domain/src/fetcher/factory.js

```diff
@@ -0,0 +1,6 @@
+import AxiosFetcher from './AxiosFetcher'
```

Comment:
Something to think about working with factories.
Having a single factory for all strategies forces the import of all used libraries.
For instance, imagine what will happen if we add the superAgentFetcher method, the code would be:

```js
import AxiosFetcher from './AxiosFetcher' // (imports axios package to bundle)
import SuperAgentFetcher from './SuperAgentFetcher' //(imports superagent package to bundle)

export default class FetcherFactory {

  static axiosFetcher = ({config}) =>
    new AxiosFetcher({config})

  static superAgentFetcher = ({config}) =>
    new SuperAgentFetcher({config})
}
```

Is it scalable ? Now that we are redefining part of the architecture, to me it makes sense to define separated factories.

Does it make sense ?

Reply:
I understand the Fetcher should be always a single implementation of a HttpFetcher. I agree that the naming couldn't be correct. I don't like to export this with a coupled library name as I don't want to let people to use whatever library they feel like (Axios, Superagent, fetch, petch, $.ajax,...). We should converge in the fetcher with the same library and then push forward that one.

Reply:
As I was saying it's more about defining the way to do a factory for domain more than for this package.
But if we define a way we should apply it here.

Reply:
TODO: Rename `static axiosFetcher` for `static httpFetcher`.